proposals/storage-and-retrieval-bots.md

```diff
@@ -0,0 +1,168 @@
+
+#### Confidence
+_How sure are we that this impact would be realized? Label from [this scale](https://medium.com/@nimay/inside-product-introduction-to-feature-priority-using-ice-impact-confidence-ease-and-gist-5180434e5b15)_.
+
+C = 8
+
+Nothing is certain, but it is very likely that building this tool will at a minimum enable the Filecoin Project to better understand the frequency and causes of deal failures.  
+
+And the ability of this tool to support miner reputation systems can only help increase deals that get routed to reliable miners.
+
+
+## Project definition
+#### Brief plan of attack
+
+<!--Briefly describe the milestones/steps/work needed for this project-->
+ - **Phase 1:  Retrieval Bot.**  Reads stdin describing a CID to attempt to retrieve, writes outcome of retrieval attempt to stdout.
+ - **Phase 2:  Storage Bot.**  Same idea but for storage deals.
+ - **Phase 3:  Orchestrator.**  Invokes the r-bot and s-bot programs with inputs one-by-one from a long queue of CIDs to test retrieve, or files to test store, etc.
+
+#### What does done look like?
+_What specific deliverables should completed to consider this project done?_
+
+![High level architecture](https://github.com/protocol/web3-dev-team/blob/bots-proposal/proposals/images/bot-arch.png)
+
+####  What does success look like?
+_Success means impact. How will we know we did the right thing?_
+
+ - We have a metrics dashboard (in Observable, Grafana, etc) that continuously shows the most recent deal failures, how frequently they are happening, which miners fail most/least, and similar metrics.  The impact of this should be obvious:  a clearer understanding of why and how often deals are failing on mainnet.
+ - Reputation systems emerge from ecosystem partners that use the data generated by running these bots to rank miners.  This would give Filecoin users reliable, real-time miner ranking, which does not currently exist in the ecosystem.
+
+#### Counterpoints &amp; pre-mortem
+_Why might this project be lower impact than expected? How could this project fail to complete, or fail to be successful?_
+
+ - The metrics fail to give us actionable debugging ideas
+ - Reputation systems develop their own code to capture the same miner statistics (duplication of effort)
+
+#### Alternatives
+_How might this project’s intent be realized in other ways (other than this project proposal)? What other potential solutions can address the same need?_
+
+ - [@whyrusleeping](https://github.com/whyrusleeping/)'s [Estuary](https://github.com/whyrusleeping/estuary) tool
+
+#### Dependencies/prerequisites
+<!--List any other projects that are dependencies/prerequisites for this project that is being pitched.-->
+
+ - [filecoin-project/lotus/pull/5833/
+](https://github.com/filecoin-project/lotus/pull/5833/)
+
+#### Future opportunities
+<!--What future projects/opportunities could this project enable?-->
+
+ - Miner reputation systems as discussed
```

Comment:
A retrieval bot is also a potentially important component in the ipfs<>filecoin story, as the bot is a natural location to track and index content CIDs within deals to be able to contribute to the content discovery problem. 
cc @pooja

Reply:
Updated